FTP/UiPath.FTP/SftpSession.cs

```diff
@@ -212,6 +212,52 @@ private async Task<IEnumerable<FtpObjectInfo>> GetRemoteListingAsync(string remo
             return listing;
         }
 
+        private IEnumerable<string> GetMissingDirectories(string remotePath)
+        {
+            if (string.IsNullOrWhiteSpace(remotePath))
+            {
+                throw new ArgumentNullException(nameof(remotePath));
+            }
+            List<string> missingDirectories = new List<string>();
+            // TODO: We really need to find a better alternative to custom code for unix path handling.
+            const char separator = '/';
+            string currentPath = remotePath.StartsWith(separator.ToString()) ? separator.ToString() : string.Empty;
+            foreach (string directory in remotePath.Trim(separator).Split(separator))
+            {
+                currentPath += directory + separator;
+                if (!((IFtpSession)this).DirectoryExists(currentPath))
+                {
+                    missingDirectories.Add(currentPath);
+                }
+            }
+            return missingDirectories;
+        }
+
+        private async Task<IEnumerable<string>> GetMissingDirectoriesAsync(string remotePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(remotePath))
+            {
+                throw new ArgumentNullException(nameof(remotePath));
+            }
+            if (cancellationToken == null)
+            {
+                throw new ArgumentNullException(nameof(cancellationToken));
+            }
+            List<string> missingDirectories = new List<string>();
+            // TODO: We really need to find a better alternative to custom code for unix path handling.
+            const char separator = '/';
+            string currentPath = remotePath.StartsWith(separator.ToString()) ? separator.ToString() : string.Empty;
+            foreach (string directory in remotePath.Trim(separator).Split(separator))
+            {
+                currentPath += directory + separator;
```

Comment:
String Builder

Reply:
No, I'm using the string at each iteration.